google/ggrc-core
# GGRC-7140/GGRC-7135/GGRC-7328 Mention a person in review

# Dependencies

This PR is `on hold` until the dependencies are merged:

- [x] #9414
- [x] GGRC-7288
- [x] On hold since needs clarification from the BA side

# Issue description

Currently, if user adds a comment/mention person in Request Review popup, comment is not shown in general comments section. User can see comment/mentioning person from email.
Suggest to make an improvement for Request Review popup comments as proposal works.

# Steps to test the changes

1. Open reviewable object, e.g. Regulation
2. Click Request Review
3. Leave a comment/make mentioning person and click on 'Request' screenshot-1.png
4. Look at the RESPONSES/COMMENTS
**Actual Result**: Comments from Request Review are not displayed in object's comment section 
**Expected Result**: Comments from Request Review should be displayed in object's comment section

**RESPONSES/COMMENTS section**
Add a message when comment for Request Review is added:
'Review requested with a comment: ''some comment".

1. Open reviewable object, e.g. Regulation
2. Click on Request Review and mention person in the comment (+/@user@google.com)
3. Click on Request
4. Open email
**Actual Result**: There is no email notification that user mentioned 'you' in Request Review
**Expected result:** Email notification should be received.

# Solution description

1. Add comment creation logic into post hook for review
2. Add mention a person logic into post commit hook for review
3. Refactor Proposal object and move hook logic into the model

# Sanity checklist




# PR Review checklist
